torchvision/csrc/io/decoder/gpu/decoder.cpp

```diff
@@ -138,38 +139,25 @@ int Decoder::handle_picture_display(CUVIDPARSERDISPINFO* disp_info) {
   }
 
   auto options = torch::TensorOptions().dtype(torch::kU8).device(torch::kCUDA);
-  torch::Tensor decoded_frame = torch::empty({get_frame_size()}, options);
+  torch::Tensor decoded_frame =
+      torch::empty({get_height(), get_width(), 3}, options);
   uint8_t* frame_ptr = decoded_frame.data_ptr<uint8_t>();
+  const uint8_t* const source_arr[] = {
+      (const uint8_t* const)source_frame,
+      (const uint8_t* const)(source_frame + source_pitch * ((surface_height + 1) & ~1))};
```

Comment:
I would like us to double-check this `(surface_height + 1) & ~1)` condition. Can you create some videos with odd dimensions to validate what is actually needed?

Reply:
[surface_height](https://github.com/pytorch/vision/blob/bf073e785528970e6a1605e411e4fc382d686dc7/torchvision/csrc/io/decoder/gpu/decoder.cpp#L323) is different from [luma_height](https://github.com/pytorch/vision/blob/bf073e785528970e6a1605e411e4fc382d686dc7/torchvision/csrc/io/decoder/gpu/decoder.cpp#L316) which is directly related to the video dimensions.
I can revisit this when doing code refactoring.

Reply:
My point is the `+1) & ~1` condition. I'm not sure if it's actually necessary

Reply:
luma height is aligned by 2, so the chroma offset should not be odd(chroma base address can't be odd memory location), hence the alignment.